source/index.ts

```diff
@@ -226,6 +223,7 @@ export default class PQueue<QueueType extends Queue<RunFunction, EnqueueOptionsT
 	/**
 	Adds a sync or async task to the queue. Always returns a promise.
 	*/
+	// eslint-disable-next-line @typescript-eslint/prefer-readonly-parameter-types
 	async add<TaskResultType>(fn: Task<TaskResultType>, options: Partial<EnqueueOptionsType> = {}): Promise<TaskResultType> {
```

Comment:
I think it's tricky to use `Readonly` types here because the `{}` assignment.

Reply:
How so? Can you elaborate? If it's not possible to use readonly here, it sounds like a the linter rule should be adjusted to ignore this case.

Reply:
If you type `options` as 

```
Readonly<Partial<EnqueueOptionsType
```

Then this type error is returned:

```
Type '{}' is not assignable to type 'Readonly<Partial<EnqueueOptionsType>>'.ts(2322)
```

And the linter _does_ want `options` to be Readonly afaict:

```
source/index.ts:229:58
  ✖  229:58  Parameter should be a read only type
```

Not an expert, but this indeed could be a linting issue I think. We shouldn't require options to be readonly in this case.

Reply:
Let me know what you'd prefer in this case. Happy to look into that linting thing too if it's confirmed an issue.

Reply:
If TypeScript doesn't allow it, it's a linter bug. Can you open an issue here: https://github.com/typescript-eslint/typescript-eslint ?